base/deprecated.jl

```diff
@@ -148,8 +148,6 @@ end
 @deprecate inf{T<:AbstractFloat}(::Type{T})  convert(T,Inf)
 @deprecate nan{T<:AbstractFloat}(::Type{T})  convert(T,NaN)
 
-@deprecate_binding String AbstractString
```

Comment:
So this is 0.6 material?

(Unless we use a different name for the native string type)